pkg/k8s/serviceCatalog.go

```diff
@@ -222,7 +226,11 @@ func (sc *serviceCatalogClient) BindToService(bindableService, targetSvcName str
 	if err != nil {
 		return errors.Wrap(err, "failed to create binding. Request error occurred")
 	}
-	defer res.Body.Close()
+	defer func() {
```

Comment:
Why are you doing this as deferred funcs here, and as a deferred call to a closeBody function in the httpClient pkg? I think we should do it the same way - whichever way that is - every where.

Reply:
had issues with circular imports. Can change to use the anon func everywhere

Reply:
actually as mentioned the httpclient func isnt actually used. Have removed it now